src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1407,10 +1406,10 @@ protected override BoundExpression VisitExpressionWithoutStackGuard(BoundExpress
             Debug.Assert((object)resultType.Type != _invalidType.Type);
             Debug.Assert(AreCloseEnough(resultType.Type, node.Type));
 #endif
-            if (node.IsSuppressed || node.HasAnyErrors || !wasReachable)
+            if (node.IsSuppressed || node.HasAnyErrors || !IsReachable())
             {
                 resultType = resultType.WithNotNullState();
-                SetResult(resultType, LvalueResultType);
+                SetResult(resultType, TypeWithAnnotations.Create(LvalueResultType.Type));
             }
```

Comment:
Suppression and reachability should not affect the nullable annotation.

Reply:
I've reverted this change to move forward.